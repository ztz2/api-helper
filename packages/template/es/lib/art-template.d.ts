import artTemplate from './utils/template/art-template-web.js';
export default artTemplate;
