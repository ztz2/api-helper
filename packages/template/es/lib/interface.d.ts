import PrettierrcOptions from '@api-helper/cli/lib/tools/prettierrc-options';
export declare class DocumentConfig {
    id: string;
    title: string;
    url: string;
    type: string;
    auth: {
        username: string;
        password: string;
    };
    authToken: string;
    dataKey: string;
    /** 以下属性是基于项目级别的通用配置项 - 开始 * */
    exportFileId: string;
    exportFilePath: string;
    apiTplId: string;
    modelTplId: string;
    headCodeText: string;
    onlyApiFunc: boolean;
    maxlength: number;
    placeholder: boolean;
    grid: boolean;
    generateLabel: boolean;
    prettierrcOptions: PrettierrcOptions;
    /** 以下属性是基于项目级别的通用配置项 - 结束 * */
    constructor(id?: string | Partial<DocumentConfig>);
}
export declare class Template {
    label: string;
    value: string;
    content: string;
    builtIn: boolean;
    formatCodeExtension: "" | ".ts" | ".tsx" | ".js" | ".jsx" | ".vue" | ".html" | ".scss" | ".less" | ".css" | ".json" | ".java";
    constructor(options?: Partial<Template>);
}
export declare class FileDirectory {
    label: string;
    value: string;
    isFolder: boolean;
    apiId: string;
    modelTplId: string;
    modelTplIndex: number;
    templateContent: string;
    children: FileDirectory[];
    constructor(options?: Partial<FileDirectory>);
}
export declare class ExportFile {
    label: string;
    value: string;
    builtIn: boolean;
    formatCodeExtension: "" | ".ts" | ".tsx" | ".js" | ".jsx" | ".vue" | ".html" | ".scss" | ".less" | ".css" | ".json" | ".java";
    fileDirectory: FileDirectory[];
    exportFilePath: string;
    constructor(options?: Partial<FileDirectory>);
}
export declare function createTemplate(id?: string | Partial<DocumentConfig>): DocumentConfig;
export declare function createProjectConfig(id?: string | Partial<DocumentConfig>): DocumentConfig;
export declare function createFileDirectory(options?: Partial<FileDirectory>): FileDirectory;
export declare function createExportFile(options?: Partial<FileDirectory>): ExportFile;
export interface ChangeCase {
    /**
     * @example
     * changeCase.camelCase('test string') // => 'testString'
     */
    camelCase: (value: string) => string;
    /**
     * @example
     * changeCase.constantCase('test string') // => 'TEST_STRING'
     */
    constantCase: (value: string) => string;
    /**
     * @example
     * changeCase.dotCase('test string') // => 'test.string'
     */
    dotCase: (value: string) => string;
    /**
     * @example
     * changeCase.headerCase('test string') // => 'Test-String'
     */
    headerCase: (value: string) => string;
    /**
     * @example
     * changeCase.lowerCase('TEST STRING') // => 'test string'
     */
    lowerCase: (value: string) => string;
    /**
     * @example
     * changeCase.lowerCaseFirst('TEST') // => 'tEST'
     */
    lowerCaseFirst: (value: string) => string;
    /**
     * @example
     * changeCase.paramCase('test string') // => 'test-string'
     */
    paramCase: (value: string) => string;
    /**
     * @example
     * changeCase.pascalCase('test string') // => 'TestString'
     */
    pascalCase: (value: string) => string;
    /**
     * @example
     * changeCase.pathCase('test string') // => 'test/string'
     */
    pathCase: (value: string) => string;
    /**
     * @example
     * changeCase.sentenceCase('testString') // => 'Test string'
     */
    sentenceCase: (value: string) => string;
    /**
     * @example
     * changeCase.snakeCase('test string') // => 'test_string'
     */
    snakeCase: (value: string) => string;
    /**
     * @example
     * changeCase.swapCase('Test String') // => 'tEST sTRING'
     */
    swapCase: (value: string) => string;
    /**
     * @example
     * changeCase.titleCase('a simple test') // => 'A Simple Test'
     */
    titleCase: (value: string) => string;
    /**
     * @example
     * changeCase.upperCase('test string') // => 'TEST STRING'
     */
    upperCase: (value: string) => string;
    /**
     * @example
     * changeCase.upperCaseFirst('test') // => 'Test'
     */
    upperCaseFirst: (value: string) => string;
}
