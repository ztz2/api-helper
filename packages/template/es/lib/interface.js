var __values = (this && this.__values) || function(o) {
    var s = typeof Symbol === "function" && Symbol.iterator, m = s && o[s], i = 0;
    if (m) return m.call(o);
    if (o && typeof o.length === "number") return {
        next: function () {
            if (o && i >= o.length) o = void 0;
            return { value: o && o[i++], done: !o };
        }
    };
    throw new TypeError(s ? "Object is not iterable." : "Symbol.iterator is not defined.");
};
var __read = (this && this.__read) || function (o, n) {
    var m = typeof Symbol === "function" && o[Symbol.iterator];
    if (!m) return o;
    var i = m.call(o), r, ar = [], e;
    try {
        while ((n === void 0 || n-- > 0) && !(r = i.next()).done) ar.push(r.value);
    }
    catch (error) { e = { error: error }; }
    finally {
        try {
            if (r && !r.done && (m = i["return"])) m.call(i);
        }
        finally { if (e) throw e.error; }
    }
    return ar;
};
import { merge } from 'lodash';
import { checkType } from '@api-helper/core/lib/utils/util';
import PrettierrcOptions from '@api-helper/cli/lib/tools/prettierrc-options';
import { DEFAULT_SELECT_API_TPL_ID, DEFAULT_SELECT_MODEL_TPL_ID, } from './constants';
var DocumentConfig = /** @class */ (function () {
    /** 以下属性是基于项目级别的通用配置项 - 结束 * */
    function DocumentConfig(id) {
        this.id = '';
        // 项目名称
        this.title = '';
        // 文档地址
        this.url = 'http://localhost:3210/app/swagger/mock';
        // 文档类型，默认可以解析 swagger
        this.type = 'swagger';
        // 访问文档可能需要认证信息，http auth验证方式
        this.auth = {
            username: '',
            password: '',
        };
        // 访问文档可能需要认证信息，通过使用token访问
        this.authToken = '';
        // 获取数据的key，body[dataKey]
        this.dataKey = '';
        /** 以下属性是基于项目级别的通用配置项 - 开始 * */
        // 选择的文件模块ID
        this.exportFileId = '';
        // 文件模块导出路径
        this.exportFilePath = '';
        // API选择的模版ID
        this.apiTplId = DEFAULT_SELECT_API_TPL_ID;
        // 模型选择的模版ID
        this.modelTplId = DEFAULT_SELECT_MODEL_TPL_ID;
        // 头部代码
        this.headCodeText = 'import request from \'@/api/request\';\n';
        // 只生成API函数
        this.onlyApiFunc = false;
        // 输入框属性：maxlength
        this.maxlength = 128;
        // 输入框属性：placeholder
        this.placeholder = true;
        // 是否使用栅格布局
        this.grid = false;
        // Form表单项是否生成label
        this.generateLabel = false;
        // prettier配置对象
        this.prettierrcOptions = new PrettierrcOptions();
        if (typeof id === 'string') {
            this.id = id;
        }
        if (Object.prototype.toString.call(id) === '[object Object]') {
            merge(this, id);
        }
    }
    return DocumentConfig;
}());
export { DocumentConfig };
var Template = /** @class */ (function () {
    function Template(options) {
        // 模板名称
        this.label = '';
        // 唯一ID
        this.value = '';
        // 模板内容
        this.content = '';
        // 是否为内置模版
        this.builtIn = false;
        // 格式化代码的文件后缀名
        this.formatCodeExtension = '.ts';
        checkType(options, 'Object') && merge(this, options);
    }
    return Template;
}());
export { Template };
var FileDirectory = /** @class */ (function () {
    function FileDirectory(options) {
        var e_1, _a;
        // 文件(夹)名称
        this.label = '';
        // 文件(夹)Id
        this.value = '';
        // 是否为文件夹
        this.isFolder = false;
        // 关联的apiId
        this.apiId = '';
        // 关联的模板id
        this.modelTplId = '';
        // 关联的模板内容下标
        this.modelTplIndex = 0;
        // 模版生成之后的内容
        this.templateContent = '';
        this.children = [];
        if (options && Object.prototype.toString.call(options) === '[object Object]') {
            try {
                for (var _b = __values(Object.entries(options)), _c = _b.next(); !_c.done; _c = _b.next()) {
                    var _d = __read(_c.value, 2), k = _d[0], v = _d[1];
                    // @ts-ignore
                    this[k] = v;
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) _a.call(_b);
                }
                finally { if (e_1) throw e_1.error; }
            }
        }
    }
    return FileDirectory;
}());
export { FileDirectory };
var ExportFile = /** @class */ (function () {
    function ExportFile(options) {
        var e_2, _a;
        // 模板名称
        this.label = '';
        // 唯一ID
        this.value = '';
        // 是否为内置模版
        this.builtIn = false;
        // 格式化代码的文件后缀名
        this.formatCodeExtension = '.ts';
        // 模板内容
        this.fileDirectory = [];
        // 文件模块导出路径
        this.exportFilePath = '';
        if (options && Object.prototype.toString.call(options) === '[object Object]') {
            try {
                for (var _b = __values(Object.entries(options)), _c = _b.next(); !_c.done; _c = _b.next()) {
                    var _d = __read(_c.value, 2), k = _d[0], v = _d[1];
                    // @ts-ignore
                    this[k] = v;
                }
            }
            catch (e_2_1) { e_2 = { error: e_2_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) _a.call(_b);
                }
                finally { if (e_2) throw e_2.error; }
            }
        }
    }
    return ExportFile;
}());
export { ExportFile };
export function createTemplate(id) {
    return new DocumentConfig(id);
}
export function createProjectConfig(id) {
    return new DocumentConfig(id);
}
export function createFileDirectory(options) {
    return new FileDirectory(options);
}
export function createExportFile(options) {
    return new ExportFile(options);
}
