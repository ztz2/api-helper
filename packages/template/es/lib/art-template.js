var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
// @ts-ignore
import artTemplate from '../lib/utils/template/art-template-web.js';
// @ts-ignore
artTemplate.defaults.escape = false;
artTemplate.defaults.minimize = false;
// 新增模板《》语法的界定符规则
artTemplate.defaults.rules[2] = __assign(__assign({}, artTemplate.defaults.rules[1]), { test: /《([@#]?)[ \t]*(\/?)([\w\W]*?)[ \t]*》/ });
artTemplate.defaults.imports.renderCharacterByBoolean = function (booleanStatus, trueValue, falseValue) {
    if (booleanStatus === void 0) { booleanStatus = true; }
    if (trueValue === void 0) { trueValue = ''; }
    if (falseValue === void 0) { falseValue = ''; }
    return booleanStatus ? trueValue : falseValue;
};
artTemplate.defaults.imports.renderTemplateLiteral = function (booleanStatus, defaultValue) {
    if (booleanStatus === void 0) { booleanStatus = true; }
    if (defaultValue === void 0) { defaultValue = '\''; }
    return booleanStatus ? '`' : defaultValue;
};
export default artTemplate;
