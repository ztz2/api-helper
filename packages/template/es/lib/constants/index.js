export var API_CUSTOM_TEMPLATE_ID = 'API自定义分组ID_ea18930468d9402e';
export var MODEL_CUSTOM_TEMPLATE_ID = '模型自定义分组ID_6f3f32d6b404aaf9';
export var DEFAULT_SELECT_API_TPL_ID = 'typescript_api_76ea205dba9f3dfb';
export var DEFAULT_SELECT_MODEL_TPL_ID = 'javascript_map_841d8ffc9b34d1d5';
