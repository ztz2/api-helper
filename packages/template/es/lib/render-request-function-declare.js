import * as _changeCase from 'change-case';
import { getSchema } from '@api-helper/core/lib/helpers';
import { renderRequestFunctionName, renderRequestFunctionComment, } from '../lib/render-request-function';
import artTemplate from '../lib/art-template';
import { renderInterfaceName } from '../lib/render-interface';
import { isEmptySchema } from '../lib/utils/util';
export function renderRequestFunctionDeclare(api, options) {
    var _a, _b, _c, _d, _e, _f, _g, _h, _j, _k, _l, _m, _o, _p, _q, _r;
    if (options === void 0) { options = {}; }
    if (!api) {
        return '';
    }
    var dataKey = options === null || options === void 0 ? void 0 : options.dataKey;
    var isEmptyRequestData = isEmptySchema(api.requestDataSchema);
    var onRenderRequestFunctionName = (_a = options === null || options === void 0 ? void 0 : options.onRenderRequestFunctionName) !== null && _a !== void 0 ? _a : renderRequestFunctionName;
    var onRenderInterfaceName = (_b = options === null || options === void 0 ? void 0 : options.onRenderInterfaceName) !== null && _b !== void 0 ? _b : renderInterfaceName;
    var responseDataSchema = dataKey ? getSchema(api.responseDataSchema, dataKey) : api.responseDataSchema;
    var headers = (_h = (_g = (_f = (_e = (_d = (_c = api.headers) === null || _c === void 0 ? void 0 : _c.params) === null || _d === void 0 ? void 0 : _d.map) === null || _e === void 0 ? void 0 : _e.call(_d, function (item) { return item.keyName; })) === null || _f === void 0 ? void 0 : _f.filter) === null || _g === void 0 ? void 0 : _g.call(_f, Boolean)) !== null && _h !== void 0 ? _h : [];
    var cookies = (_p = (_o = (_m = (_l = (_k = (_j = api.cookies) === null || _j === void 0 ? void 0 : _j.params) === null || _k === void 0 ? void 0 : _k.map) === null || _l === void 0 ? void 0 : _l.call(_k, function (item) { return item.keyName; })) === null || _m === void 0 ? void 0 : _m.filter) === null || _o === void 0 ? void 0 : _o.call(_m, Boolean)) !== null && _p !== void 0 ? _p : [];
    var templateTenderParams = {
        api: api,
        options: options,
        isEmptyRequestData: isEmptyRequestData,
        commentCode: renderRequestFunctionComment(api),
        formDataKeyNameListStr: JSON.stringify(api.formDataKeyNameList),
        pathParamKeyNameListStr: JSON.stringify(api.pathParamKeyNameList),
        queryStringKeyNameListStr: JSON.stringify(api.queryStringKeyNameList),
        headersStr: JSON.stringify(headers),
        cookiesStr: JSON.stringify(cookies),
        requestContentTypeStr: JSON.stringify((_q = api.requestContentType) !== null && _q !== void 0 ? _q : []),
        responseContentTypeStr: JSON.stringify((_r = api.responseContentType) !== null && _r !== void 0 ? _r : []),
        requestFunctionName: onRenderRequestFunctionName(api, {
            changeCase: _changeCase,
        }),
        requestDataInterfaceName: onRenderInterfaceName(api, {
            paramType: 'request',
            changeCase: _changeCase,
            schema: api.requestDataSchema,
        }),
        requestExtraDataInterfaceName: onRenderInterfaceName(api, {
            isExtraData: true,
            paramType: 'request',
            changeCase: _changeCase,
            schema: api.requestExtraDataSchema,
        }),
        responseDataInterfaceName: onRenderInterfaceName(api, {
            paramType: 'response',
            changeCase: _changeCase,
            schema: responseDataSchema,
        }),
    };
    var code = artTemplate.render("\u300Aif commentCode\u300B\u300AcommentCode\u300B\n\u300A/if\u300Bdeclare const \u300ArequestFunctionName\u300B: {\n  (data\u300Aif isEmptyRequestData\u300B?\u300A/if\u300B: \u300ArequestDataInterfaceName\u300B, extraData?: \u300Aif api.requestExtraDataSchema\u300B\u300ArequestExtraDataInterfaceName\u300B\u300Aelse\u300Bunknown\u300A/if\u300B, ...args: CurrentRequestFunctionRestArgsType): Promise<\u300AresponseDataInterfaceName\u300B>;\n  requestConfig: {\n    path: '\u300Aapi.path\u300B',\n    method: '\u300Aapi.method.toLowerCase()\u300B',\u300Aif options.genHeaders\u300B\n    headers: \u300AheadersStr\u300B,\u300A/if\u300B\u300Aif options.genCookies\u300B\n    cookies: \u300AcookiesStr\u300B,\u300A/if\u300B\u300Aif options.genRequestContentType\u300B\n    requestContentType: \u300ArequestContentTypeStr\u300B,\u300A/if\u300B\u300Aif options.genResponseContentType\u300B\n    responseContentType: \u300AresponseContentTypeStr\u300B,\u300A/if\u300B\n    formDataKeyNameList: \u300AformDataKeyNameListStr\u300B,\n    pathParamKeyNameList: \u300ApathParamKeyNameListStr\u300B,\n    queryStringKeyNameList: \u300AqueryStringKeyNameListStr\u300B\n  }\n};", templateTenderParams);
    return code;
}
