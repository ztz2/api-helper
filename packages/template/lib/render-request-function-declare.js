"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.renderRequestFunctionDeclare = void 0;
var _changeCase = __importStar(require("change-case"));
var helpers_1 = require("@api-helper/core/lib/helpers");
var render_request_function_1 = require("../lib/render-request-function");
var art_template_1 = __importDefault(require("../lib/art-template"));
var render_interface_1 = require("../lib/render-interface");
var util_1 = require("../lib/utils/util");
function renderRequestFunctionDeclare(api, options) {
    if (!api) {
        return '';
    }
    var dataKey = options === null || options === void 0 ? void 0 : options.dataKey;
    var isEmptyRequestData = (0, util_1.isEmptySchema)(api.requestDataSchema);
    var onRenderRequestFunctionName = (options && options.onRenderRequestFunctionName) ? options.onRenderRequestFunctionName : render_request_function_1.renderRequestFunctionName;
    var onRenderInterfaceName = (options && options.onRenderInterfaceName) ? options.onRenderInterfaceName : render_interface_1.renderInterfaceName;
    var responseDataSchema = dataKey ? (0, helpers_1.getSchema)(api.responseDataSchema, dataKey) : api.responseDataSchema;
    var templateTenderParams = {
        api: api,
        isEmptyRequestData: isEmptyRequestData,
        commentCode: (0, render_request_function_1.renderRequestFunctionComment)(api),
        formDataKeyNameListStr: JSON.stringify(api.formDataKeyNameList),
        pathParamKeyNameListStr: JSON.stringify(api.pathParamKeyNameList),
        queryStringKeyNameListStr: JSON.stringify(api.queryStringKeyNameList),
        requestFunctionName: onRenderRequestFunctionName(api, {
            changeCase: _changeCase,
        }),
        requestDataInterfaceName: onRenderInterfaceName(api, {
            paramType: 'request',
            changeCase: _changeCase,
            schema: api.requestDataSchema,
        }),
        requestExtraDataInterfaceName: onRenderInterfaceName(api, {
            isExtraData: true,
            paramType: 'request',
            changeCase: _changeCase,
            schema: api.requestExtraDataSchema,
        }),
        responseDataInterfaceName: onRenderInterfaceName(api, {
            paramType: 'response',
            changeCase: _changeCase,
            schema: responseDataSchema,
        }),
    };
    var code = art_template_1.default.render("\u300Aif commentCode\u300B\u300AcommentCode\u300B\n\u300A/if\u300Bdeclare const \u300ArequestFunctionName\u300B: {\n  (data\u300Aif isEmptyRequestData\u300B?\u300A/if\u300B: \u300ArequestDataInterfaceName\u300B, extraData?: \u300Aif api.requestExtraDataSchema\u300B\u300ArequestExtraDataInterfaceName\u300B\u300Aelse\u300Bunknown\u300A/if\u300B, ...args: CurrentRequestFunctionRestArgsType): Promise<\u300AresponseDataInterfaceName\u300B>;\n  requestConfig: {\n    path: '\u300Aapi.path\u300B',\n    method: '\u300Aapi.method.toLowerCase()\u300B',\n    formDataKeyNameList: \u300AformDataKeyNameListStr\u300B,\n    pathParamKeyNameList: \u300ApathParamKeyNameListStr\u300B,\n    queryStringKeyNameList: \u300AqueryStringKeyNameListStr\u300B\n  }\n};", templateTenderParams);
    return code;
}
exports.renderRequestFunctionDeclare = renderRequestFunctionDeclare;
