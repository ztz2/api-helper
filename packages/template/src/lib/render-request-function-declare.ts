import * as _changeCase from 'change-case';
import { APIHelper } from '@api-helper/core/lib/types';
import { getSchema } from '@api-helper/core/lib/helpers';

import {
  renderRequestFunctionName,
  renderRequestFunctionComment,
} from '@/lib/render-request-function';
import artTemplate from '@/lib/art-template';
import { renderInterfaceName } from '@/lib/render-interface';
import { isEmptySchema } from '@/lib/utils/util';

export function renderRequestFunctionDeclare(
  api: APIHelper.API,
  options: {
    dataKey?: string | undefined;
    genHeaders?: boolean;
    genCookies?: boolean;
    genRequestContentType?: boolean;
    genResponseContentType?: boolean;
    onRenderInterfaceName?: typeof renderInterfaceName;
    onRenderRequestFunctionName?: typeof renderRequestFunctionName;
  } = {}
) {
  if (!api) {
    return '';
  }
  const dataKey = options?.dataKey;
  const isEmptyRequestData = isEmptySchema(api.requestDataSchema);
  const onRenderRequestFunctionName = options?.onRenderRequestFunctionName ?? renderRequestFunctionName;
  const onRenderInterfaceName = options?.onRenderInterfaceName ?? renderInterfaceName;
  const responseDataSchema = dataKey ? getSchema(api.responseDataSchema, dataKey) : api.responseDataSchema;
  const headers = api.headers?.params?.map?.((item) => item.keyName)?.filter?.(Boolean) ?? [];
  const cookies = api.cookies?.params?.map?.((item) =>  item.keyName)?.filter?.(Boolean) ?? [];
  const templateTenderParams = {
    api,
    options,
    isEmptyRequestData,
    commentCode: renderRequestFunctionComment(api),
    formDataKeyNameListStr: JSON.stringify(api.formDataKeyNameList),
    pathParamKeyNameListStr: JSON.stringify(api.pathParamKeyNameList),
    queryStringKeyNameListStr: JSON.stringify(api.queryStringKeyNameList),
    headersStr: JSON.stringify(headers),
    cookiesStr: JSON.stringify(cookies),
    requestContentTypeStr: JSON.stringify(api.requestContentType ?? []),
    responseContentTypeStr: JSON.stringify(api.responseContentType ?? []),
    requestFunctionName: onRenderRequestFunctionName(api, {
      changeCase: _changeCase,
    }),
    requestDataInterfaceName: onRenderInterfaceName(api, {
      paramType: 'request',
      changeCase: _changeCase,
      schema: api.requestDataSchema as APIHelper.Schema,
    }),
    requestExtraDataInterfaceName: onRenderInterfaceName(api, {
      isExtraData: true,
      paramType: 'request',
      changeCase: _changeCase,
      schema: api.requestExtraDataSchema as APIHelper.Schema,
    }),
    responseDataInterfaceName: onRenderInterfaceName(api, {
      paramType: 'response',
      changeCase: _changeCase,
      schema: responseDataSchema,
    }),
  };

  const code = artTemplate.render(
    `《if commentCode》《commentCode》
《/if》declare const 《requestFunctionName》: {
  (data《if isEmptyRequestData》?《/if》: 《requestDataInterfaceName》, extraData?: 《if api.requestExtraDataSchema》《requestExtraDataInterfaceName》《else》unknown《/if》, ...args: CurrentRequestFunctionRestArgsType): Promise<《responseDataInterfaceName》>;
  requestConfig: {
    path: '《api.path》',
    method: '《api.method.toLowerCase()》',《if options.genHeaders》
    headers: 《headersStr》,《/if》《if options.genCookies》
    cookies: 《cookiesStr》,《/if》《if options.genRequestContentType》
    requestContentType: 《requestContentTypeStr》,《/if》《if options.genResponseContentType》
    responseContentType: 《responseContentTypeStr》,《/if》
    formDataKeyNameList: 《formDataKeyNameListStr》,
    pathParamKeyNameList: 《pathParamKeyNameListStr》,
    queryStringKeyNameList: 《queryStringKeyNameListStr》
  }
};`, templateTenderParams);

  return code;
}
