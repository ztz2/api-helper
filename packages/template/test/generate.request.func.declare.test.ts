import { join } from 'path';
import ParserSwagger from '@api-helper/core/es/lib/parser/parser-swagger';

import { readJsonFile } from './utils/file';
import { renderRequestFunctionDeclare } from '../src/lib/render-request-function-declare';

describe('生成接口函数测试', () => {
  test('OpenAPI-2.0生成接口函数测试', async () => {
    const openAPIDocumentJSON = await readJsonFile(join(__dirname, '../../core/test/resources/open-api/v2.0/petstore.json'));
    const openAPIDocumentList = await new ParserSwagger(false).parser([openAPIDocumentJSON]);
    const openAPIDocument = openAPIDocumentList[0];
    const category = openAPIDocument.categoryList[1];
    const codeBox: string[] = [];
    for (const api of category.apiList) {
      codeBox.push(renderRequestFunctionDeclare(api));
    }
    expect(
      codeBox.join('\n'),
    ).toMatchSnapshot('OpenAPI-2.0生成接口函数测试');
  });

  test('OpenAPI-3.0生成接口函数测试', async () => {
    const openAPIDocumentJSON = await readJsonFile(join(__dirname, '../../core/test/resources/open-api/v3.0/petstore.json'));
    const openAPIDocumentList = await new ParserSwagger(false).parser([openAPIDocumentJSON]);
    const openAPIDocument = openAPIDocumentList[0];
    const category = openAPIDocument.categoryList[1];
    const codeBox: string[] = [];
    for (const api of category.apiList) {
      codeBox.push(renderRequestFunctionDeclare(api));
    }
    expect(
      codeBox.join('\n'),
    ).toMatchSnapshot('OpenAPI-3.0生成接口函数测试');
  });
});
