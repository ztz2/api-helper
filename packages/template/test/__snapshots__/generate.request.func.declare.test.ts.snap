// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`生成接口函数测试 OpenAPI-2.0生成接口函数测试: OpenAPI-2.0生成接口函数测试 1`] = `
"/**
   * @description Returns pet inventories by status、Returns a map of status codes to quantities
   * @url [ GET ] /v2/store/inventory
   */
declare const v2StoreInventoryByGet: {
  (data?: V2StoreInventoryRequestByGet, extraData?: unknown, ...args: CurrentRequestFunctionRestArgsType): Promise<V2StoreInventoryResponseByGet>;
  requestConfig: {
    path: '/v2/store/inventory',
    method: 'get',
    formDataKeyNameList: [],
    pathParamKeyNameList: [],
    queryStringKeyNameList: []
  }
};
/**
   * @description Place an order for a pet
   * @url [ POST ] /v2/store/order
   */
declare const v2StoreOrderByPost: {
  (data: V2StoreOrderRequestByPost, extraData?: unknown, ...args: CurrentRequestFunctionRestArgsType): Promise<V2StoreOrderResponseByPost>;
  requestConfig: {
    path: '/v2/store/order',
    method: 'post',
    formDataKeyNameList: [],
    pathParamKeyNameList: [],
    queryStringKeyNameList: []
  }
};
/**
   * @description Find purchase order by ID、For valid response try integer IDs with value >= 1 and <= 10. Other values will generated exceptions
   * @url [ GET ] /v2/store/order/{orderId}
   */
declare const v2StoreOrderOrderIdByGet: {
  (data: V2StoreOrderOrderIdRequestByGet, extraData?: unknown, ...args: CurrentRequestFunctionRestArgsType): Promise<V2StoreOrderOrderIdResponseByGet>;
  requestConfig: {
    path: '/v2/store/order/{orderId}',
    method: 'get',
    formDataKeyNameList: [],
    pathParamKeyNameList: ["orderId"],
    queryStringKeyNameList: []
  }
};
/**
   * @description Delete purchase order by ID、For valid response try integer IDs with positive integer value. Negative or non-integer values will generate API errors
   * @url [ DELETE ] /v2/store/order/{orderId}
   */
declare const v2StoreOrderOrderIdByDelete: {
  (data: V2StoreOrderOrderIdRequestByDelete, extraData?: unknown, ...args: CurrentRequestFunctionRestArgsType): Promise<V2StoreOrderOrderIdResponseByDelete>;
  requestConfig: {
    path: '/v2/store/order/{orderId}',
    method: 'delete',
    formDataKeyNameList: [],
    pathParamKeyNameList: ["orderId"],
    queryStringKeyNameList: []
  }
};"
`;

exports[`生成接口函数测试 OpenAPI-3.0生成接口函数测试: OpenAPI-3.0生成接口函数测试 1`] = `
"/**
   * @description Returns pet inventories by status、Returns a map of status codes to quantities
   * @url [ GET ] /api/v3/store/inventory
   */
declare const apiV3StoreInventoryByGet: {
  (data?: ApiV3StoreInventoryRequestByGet, extraData?: unknown, ...args: CurrentRequestFunctionRestArgsType): Promise<ApiV3StoreInventoryResponseByGet>;
  requestConfig: {
    path: '/api/v3/store/inventory',
    method: 'get',
    formDataKeyNameList: [],
    pathParamKeyNameList: [],
    queryStringKeyNameList: []
  }
};
/**
   * @description Place an order for a pet、Place a new order in the store
   * @url [ POST ] /api/v3/store/order
   */
declare const apiV3StoreOrderByPost: {
  (data: ApiV3StoreOrderRequestByPost, extraData?: unknown, ...args: CurrentRequestFunctionRestArgsType): Promise<ApiV3StoreOrderResponseByPost>;
  requestConfig: {
    path: '/api/v3/store/order',
    method: 'post',
    formDataKeyNameList: [],
    pathParamKeyNameList: [],
    queryStringKeyNameList: ["id","petId","quantity","shipDate","status","complete"]
  }
};
/**
   * @description Find purchase order by ID、For valid response try integer IDs with value <= 5 or > 10. Other values will generate exceptions.
   * @url [ GET ] /api/v3/store/order/{orderId}
   */
declare const apiV3StoreOrderOrderIdByGet: {
  (data: ApiV3StoreOrderOrderIdRequestByGet, extraData?: unknown, ...args: CurrentRequestFunctionRestArgsType): Promise<ApiV3StoreOrderOrderIdResponseByGet>;
  requestConfig: {
    path: '/api/v3/store/order/{orderId}',
    method: 'get',
    formDataKeyNameList: [],
    pathParamKeyNameList: ["orderId"],
    queryStringKeyNameList: []
  }
};
/**
   * @description Delete purchase order by ID、For valid response try integer IDs with value < 1000. Anything above 1000 or nonintegers will generate API errors
   * @url [ DELETE ] /api/v3/store/order/{orderId}
   */
declare const apiV3StoreOrderOrderIdByDelete: {
  (data: ApiV3StoreOrderOrderIdRequestByDelete, extraData?: unknown, ...args: CurrentRequestFunctionRestArgsType): Promise<ApiV3StoreOrderOrderIdResponseByDelete>;
  requestConfig: {
    path: '/api/v3/store/order/{orderId}',
    method: 'delete',
    formDataKeyNameList: [],
    pathParamKeyNameList: ["orderId"],
    queryStringKeyNameList: []
  }
};"
`;
